### DISCUSSION TITLE
Can't connect with Jetbrains Datagrip

### DISCUSSION THREAD

**Discussion by Ncage1974** [planetscale/discussion#568] [Q&A] [answered]
![image](https://github.com/planetscale/discussion/assets/1213549/c7075ba6-6e42-430f-a6ec-05b33dddf8e4)![image](https://github.com/planetscale/discussion/assets/1213549/e606a0a8-088a-4c46-9f1c-8fa368ef91b6)
![image](https://github.com/planetscale/discussion/assets/1213549/e5e7619d-0f99-4684-ad15-e296f772936e)

I have tried a wide variety of options &, in generally, this is the error i'm getting. I tried with & without the default port (3306). I tried using port 443 (SSL right). I tried with and without the DB Name. I looked at the example how to connect with JDBC (its using java behind the scenes & tried adding `?sslMode=VERIFY_IDENTITY` to the end of the URL. 

In the "Connection Type" in the general tab you can see its set at default. I tried the two other types: "Unix Sockets" & "URL Only". The closes i've came, i think, is when i chose port 443 but it spins forever & never completes....all the other options fail immediately with the dialog in the screenshot. 

Edit:

Just tried mysql workbench & can't connect with that either

Edit: 
FYI I tried connecting with a variety of other products (thinking maybe we were doing something wrong) we are comparing to planetscale which includes cockroachdb & neon & didn't have any issues connecting.

**Comment by orware** [comment, marked as answer]
Hello @Ncage1974,

The error shown about "Connection refused" likely indicates you are pointing to a location, localhost over port 3306 in this case, that may not have anything running on it, therefore the connection is going to be automatically refused.

Are you also using the `pscale connect` command in conjunction with the example screenshots you provided?

I ask because the hostname you have shown in the screenshot is localhost, which might imply you were using the [pscale connect option](https://planetscale.com/docs/reference/connect) for connecting to your database, but in that situation it handles the behind the scenes SSL/TLS setup for the connection so you don't use SSL when configuring a connection in that case ([Example](https://planetscale.com/docs/tutorials/connect-any-application#option-2-connect-using-the-planetscale-proxy)). 

If you are just trying to connect to a PlanetScale database directly, then I would expect you to be using our generic hostname in most cases: `aws.connect.psdb.cloud`.

On the General tab shown in the first screenshot, you would normally fill in the hostname with generic one I just shared, along with a username and password generated from within the PlanetScale UI and providing the database name as well.

Using the default port of 3306 is what you would need to do here since that's the port the regular MySQL protocol operates within, even if secured by SSL/TLS. We do also provide a separate HTTPS access option over port 443, but that's specifically for the [serverless JavaScript driver](https://planetscale.com/docs/tutorials/planetscale-serverless-driver) and would not work for connecting a regular MySQL client since it uses an HTTP protocol vs. the regular MySQL one.

For the SSH/SSL tab within DataGrip, checking the "Use SSL" checkbox is a good start and it does look like it will attempt to use existing certificates built in to your IDE / Java / System with the "Use truststore" option checked. 

If for some reason you receive an SSL related error in this case, you could optionally download the Let's Encrypt Root CA file directly from their site:
https://letsencrypt.org/certs/isrgrootx1.pem

And provide the path to that file for the "CA file" field.

I hope these tips are helpful, but if you continue to have difficulty please reach out to us over at https://support.planetscale.com/ if needed and we can also work with you more closely / quickly.

**Reply by Ncage1974** [reply]
Hi @orware

thanks for the reply. I was trying so many things to fix this issue. I thought i specified the correct host but i guess not. I appreciate your help & it is working now.

**Reply by orware** [reply]
Happy to hear it @Ncage1974 and thank you for confirming things are working now for you!

**Comment by Ncage1974** [comment]
FYI i hope this helps someone else. I end up going home & started to have the issue again. So i dug in deeper. Come to find out my IDS (intrusion Detection System) was blocking it:

`3.15.73.63 | 3306 | tcp | worm | trojan-activity | ET WORM Potential MySQL bot scanning for SQL server`

FYI @orware

**Reply by orware** [reply]
@Ncage1974,

Thank you for the follow-up!

That's the first time I'm seeing an IDS involved but we've definitely had a few other folks have issues connecting due to work / campus firewalls also blocking the outgoing traffic on port 3306 so this makes sense that it can also be triggered by an IDS.
